src/app/user-times/user-times.component.ts

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright 2019-present Open Networking Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Component, Input, OnInit} from '@angular/core';
+import {TsWeek, TsweeksService} from '../tsweeks.service';
+import {TsDay, TsdaysService} from '../tsdays.service';
+import {TsWeekly, TsweeklyService} from '../tsweekly.service';
+import {OAuthService} from 'angular-oauth2-oidc';
+import {generate} from 'rxjs';
+import {EMAIL_ATTR} from '../app.component';
+
+const msInDay = 24 * 60 * 60 * 1000;
+
+@Component({
+  selector: 'app-user-times',
```

Comment:
I can't see where this is being used

Reply:
Its used in the subscribe for tsweeksService